connectivedx/fuzzy-chainsaw
# Rolls color exports into custom POSTCSS lib

Replaces our colors-export gulp task(s) with custom postcss plugin.

![Wooop woop wop wop wop](https://media.giphy.com/media/wKn6stJovH0Qg/giphy.gif)

This does a few things for us:
- Removes the need for more plugin dependencies.
- Control over our colors JSON formatting.
- Removes extra watch task around exported JSON file that previously prevented build loop.

## Description
- Adds new 'styleguide' path helper
- Adds new POSTCSS export-colors lib
- Adds export-colors lib to postcss-plugins stream
- Removes color export/watch from gulp.js file
- Renames SgColorSwatch__variables.json to SgColorsSwatch__Colors.json
- Removes gulp-task required plugins from package.json

## Motivation and Context
The motivation behind this is to remove duplicate watch process on top of webpack-dev-server and inherit the colors export into the dev workflow.

## How Has This Been Tested?
- run npm or yarn dev.
- open colors page.
- edit a colors name in source/elements/variables/colors.css, see change in colors page.
- edit a colors value in source/elements/variables/colors.css, see change in colors page.
- confirm npm or yarn build, production and watch all still work

## Screenshots (if appropriate):

## Types of changes


## Checklist:
